nucypher/blockchain/economics.py

```diff
@@ -14,60 +14,166 @@
 You should have received a copy of the GNU Affero General Public License
 along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
 """
+from typing import Tuple
+
+import math
+from decimal import Decimal
+
 from math import log
 
 
-class Economics:
-    def __init__(
-            self,
-            initial_supply=10**9,
-            initial_inflation=1.0,
-            T_half=2.0,
-            T_sat=1.0,
-            small_staker_multiplier=0.5) -> None:
+class TokenEconomics:
+    """
+    Calculate parameters to use in token and escrow deployment
+    from high-level human-understandable parameters.
+
+    Formula for staking in one period:
+    (totalSupply - currentSupply) * (lockedValue / totalLockedValue) * (k1 + allLockedPeriods) / k2
+
+    K2 - Staking coefficient
+    K1 - Locked periods coefficient
+
+    if allLockedPeriods > awarded_periods then allLockedPeriods = awarded_periods
+    kappa * log(2) / halving_delay === (k1 + allLockedPeriods) / k2
+
+    ...but also...
+
+    kappa = (small_stake_multiplier + (1 - small_stake_multiplier) * min(T, T1) / T1)
+    where allLockedPeriods == min(T, T1)
+    """
+
+    # Token Denomination
+    __token_decimals = 18
```

Comment:
Should we use `decimal.getcontext().prec = __token_decimals`? Or maybe use a context manager, or whatever is appropriate.

More details here https://docs.python.org/3/library/decimal.html#decimal.getcontext

Reply:
:heavy_check_mark: